### PR TITLE
Qr lucky money

### PR DESCRIPTION
**Pull request by phuongtrinhviet**
# Ticket
https://github.com/mezonai/mmn-tx-explorer/issues/436

# General
- [x] No unnecessary files, debug logs, or commented code included.
- [x] PR has at least 1 peer approval before senior review.
- [x] I fully understand the code I submitted.
- [x] No hardcoded configuration or secrets.
- [x] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [x] A design diagram (sequence / flow) is attached or linked.
- [x] The implementation matches the design.
- [x] All major flows and decision points are represented in the diagram.
- [x] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
QR_LuckyMoney_Diagram

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [x] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [x] No obvious logic or security issues found.
- [x] Tests are sufficient for the scope of this change.

### CODE REVIEW

File: dong-service/utils/handler_helpers.go [hunk 1]
```diff
@@ -18,7 +18,12 @@ func GetUserIDFromContext(c *gin.Context) (int64, error) {
 		return 0, jwt.ErrTokenInvalidClaims
 	}
 
-	userIDStr, ok := user.(jwt.MapClaims)["user_id"].(string)
+	claims, ok := user.(jwt.MapClaims)
```
[the remark this answers is not in the document]
Comment (phuongtrinhviet) [reply dong-service/utils/handler_helpers.go]: Code cũ sẽ bị panic khi ZK middleware set user là map[string]interface{} thay vì jwt.MapClaims, tách riêng để check type an toàn trước khi access field, tránh crash runtime.

[the remark this answers is not in the document]
Comment (hieudinhngoc-ncc) [reply dong-service/utils/handler_helpers.go]: Oke, thế thì xử lý thế là đúng rồi.

**Review by hieudinhngoc-ncc — APPROVED** [review]

File: dong-service/handlers/red_envelope.go [hunk 2]
```diff
@@ -496,11 +496,142 @@ func (r *RedEnvelopeHandler) ClaimRedEnvelope(c *gin.Context) {
...
+	userAddress, ok := utils.GetAddressFromContext(c)
+	if !ok {
+		logger.Error().Msg("Address not found in context")
+		c.JSON(http.StatusUnauthorized, models.ErrorResponse(http.StatusUnauthorized, constants.ErrUnauthorized))
+		return
+	}
+
+	id := c.Query("id")
+
+	claimStatus, err := r.queueService.AttemptClaimByAddress(id, userAddress)
```
Comment (hoangdohuy) [inline dong-service/handlers/red_envelope.go]: Đoạn này mà mình vẫn dùng hàm Attempt claim cũ thì sao em nhỉ

File: dong-service/repository/red_envelope_queue.go [hunk 3]
```diff
@@ -135,3 +165,20 @@ func (s *RedEnvelopeQueueService) RollbackClaim(redEnvelopeID string, userID int
 			Msg("CRITICAL: Failed to rollback redis claim")
 	}
 }
+
+func (s *RedEnvelopeQueueService) RollbackClaimByAddress(redEnvelopeID string, address string) {
```
Comment (hoangdohuy) [inline dong-service/repository/red_envelope_queue.go]: remove this function use old function

File: dong-service/repository/red_envelope_queue.go [hunk 4]
```diff
@@ -119,6 +119,36 @@ func (s *RedEnvelopeQueueService) AttemptClaim(redEnvelopeID string, userID int6
 	}
 }
 
+func (s *RedEnvelopeQueueService) AttemptClaimByAddress(redEnvelopeID string, address string) (int, error) {
```
Comment (hoangdohuy) [inline dong-service/repository/red_envelope_queue.go]: remove this function use old function

**Review by hoangdohuy — APPROVED** [review]
LGTM
